go/pkg/client/tree.go

```diff
@@ -440,9 +440,21 @@ func (c *Client) ComputeMerkleTree(ctx context.Context, execRoot, workingDir, re
 			}
 			continue
 		}
+		if i.InputDigest != "" && i.Contents != nil {
+			return digest.Empty, nil, nil, errors.New("digest and file content cannot be provided for the same virtual input")
+		}
+		entry := uploadinfo.EntryFromBlob(i.Contents)
```

Comment:
It's worth placing this in an `else` branch to make it a bit clearer that either bytes or hash is used. It also avoids the cost of creating unnecessary structs.

Reply:
I wanted to preserve the current behaviour. We don't actually check that the Contents are not empty before creating an entry. It looks like we just check that the digest is empty before downloading https://github.com/bazelbuild/remote-apis-sdks/blob/master/go/pkg/client/cas_upload.go#L286

Reply:
The `if-else` block would only alter the source of the entry. If the digest is not specified, the entry will still be created from the content regardless of its actual value.